### PR TITLE
Skin Preview

### PR DESCRIPTION
**Pull request by danielkrupinski**
#2482

### CODE REVIEW

**Comment by wemanzoz** [comment]
LETS GOO, LETS FUCKING GOOO

**Comment by wemanzoz** [comment]
also, are you going to merge it? or we gonna paste it ourself (just askin)

**Comment by doggobot-owner** [comment]
AYYYYYYYYYYYYYYYYYYYYY

**Comment by doggobot-owner** [comment]
Hey @danielkrupinski , am i the only one whose game crashes and gives errors here:

                g_pd3dDevice->SetTexture(0, texture);
                g_pd3dDevice->SetTextureStageState(0, D3DTSS_COLOROP, texture == g_FontTexture ? D3DTOP_SELECTARG2 : D3DTOP_MODULATE);
               
      texture is not defined

File: Osiris/Hacks/SkinChanger.cpp [hunk 1]
```diff
@@ -85,8 +99,8 @@ static void initializeKits() noexcept
     kitsWeapons.reserve(itemSchema->alternateIcons.numElements);
 
     for (const auto& node : itemSchema->alternateIcons) {
-        const auto encoded = node.key;
-        kitsWeapons.emplace_back(int((encoded & 0xFFFF) >> 2), WeaponId(encoded >> 16), node.value.simpleName.buffer.memory); // https://github.com/perilouswithadollarsign/cstrike15_src/blob/f82112a2388b841d72cb62ca48ab1846dfcc11c8/game/shared/econ/econ_item_schema.cpp#L325-L329
+        if (const auto encoded = node.key; (encoded & 3) == 0)
```
Comment (rollraw) [inline Osiris/Hacks/SkinChanger.cpp:103]: love to see you figured out the right way to get it in place of all the other people still using Beaker's method

**Comment by doggobot-owner** [comment]
This code is crashing my game all of the time pl s fi x

**Pull request merged**